### PR TITLE
Add control servers

### PR DESCRIPTION
**Pull request by nahiluhmot**
@tlunter @djosborne 

Add control servers (which can have many users) with HTTP routes.

### CODE REVIEW

File: app/auth/models/control_server.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+      validate :valid_ip?
+
+      def as_json(opts = {})
+        super(opts.merge(except: %w(created_at updated_at)))
+      end
+
+      private
+
+      def valid_ip?
```
Comment (tlunter) [inline app/auth/models/control_server.rb]: Gotta hope IPv6 doesn't take off before the end of this project :D

File: app/auth/models/user.rb [hunk 2]
```diff
@@ -4,14 +4,27 @@ module Models
...
+      has_many :sessions
+      belongs_to :control_server
+
       validates :name, presence: true
       validates :email, presence: true, uniqueness: true
 
       validates :password_salt, presence: true
       validates :password_hash, presence: true
 
+      validate :control_server_associated?
```
Comment (tlunter) [inline app/auth/models/user.rb]: What happens if this isn't true? Can't save the model ever again?

**Comment by tlunter** [comment]
LGTM pending comments in other PR.

File: app/auth/models/control_server.rb (hunk 1 above)
Comment (nahiluhmot) [inline app/auth/models/control_server.rb]: Fat chance!

File: app/auth/models/user.rb (hunk 2 above)
Comment (nahiluhmot) [inline app/auth/models/user.rb]: I suppose it would -- I can't think of an instance where that would happen aside from manually deleting a control_server.

Comment (tlunter) [inline app/auth/models/user.rb]: +1

**Pull request merged**